### ISSUE TITLE
OSX problem with clang

### ISSUE THREAD

**Issue by twiecki** [conda-forge/pytensor-suite-feedstock#37]
### Comment:

```
>>micromamba install "pytensor>=2.12.0"

                                           __
          __  ______ ___  ____ _____ ___  / /_  ____ _
         / / / / __ `__ \/ __ `/ __ `__ \/ __ \/ __ `/
        / /_/ / / / / / / /_/ / / / / / / /_/ / /_/ /
       / .___/_/ /_/ /_/\__,_/_/ /_/ /_/_.___/\__,_/
      /_/

conda-forge/osx-arm64                                6.7MB @   2.7MB/s  2.9s
conda-forge/noarch                                  13.0MB @   2.8MB/s  5.2s
error    libmamba Could not solve for environment specs
    The following package could not be installed
    └─ pytensor >=2.12.0  is uninstallable because it requires
       ├─ clang_osx-arm64 15.* , which can be installed;
       └─ cxx-compiler   but there are no viable options
          ├─ cxx-compiler [1.1.2|1.1.3|1.2.0|1.3.0] would require
          │  └─ c-compiler [1.1.2 h27ca646_0|1.1.3 h27ca646_0|1.2.0 h3422bc3_0|1.3.0 h3422bc3_0], which requires
          │     └─ clang_osx-arm64 11.* , which conflicts with any installable versions previously reported;
          ├─ cxx-compiler [1.4.0|1.4.1] would require
          │  └─ c-compiler [1.4.0 hccd94f0_0|1.4.1 hccd94f0_0], which requires
          │     └─ clang_osx-arm64 12.* , which conflicts with any installable versions previously reported;
          ├─ cxx-compiler 1.4.2 would require
          │  └─ c-compiler 1.4.2 h50ab0ce_0, which requires
          │     └─ clang_osx-arm64 13.* , which conflicts with any installable versions previously reported;
          └─ cxx-compiler [1.5.0|1.5.1|1.5.2] would require
             └─ c-compiler [1.5.0 h5008568_0|1.5.1 h5008568_0|1.5.2 h5008568_0], which requires
                └─ clang_osx-arm64 14.* , which conflicts with any installable versions previously reported.
critical libmamba Could not solve for environment specs
```

**Comment by maresb** [comment]
While #36 was also an issue, this particular error is being caused by an incompatibility between
```
clang_osx-arm64 15.*
cxx-compiler
```
which is of course consistent with the error message.

I feel like we should be using `{{ compiler('cxx') }}` rather than `cxx-compiler`. A long time ago I got an error when trying to switch, which I regrettably didn't document. Now that this feedstock has evolved into something more stable I think it's time to try again. Luckily I believe that the error is covered by the tests I wrote. Unfortunately the tests only run on linux-64, osx-64 and win-64. Let's see what happens...
